### PR TITLE
perf(test): use fast ops for deno test register

### PR DESCRIPTION
**Pull request by lucacasonato**
Use fast ops for test registration. This speeds up `Deno.test` and `t.step()`
significantly (2x over Deno 1.37.0).

Depends on #20669

# `Deno.test`

```ts
for (let i = 0; i < 10000; i++) {
  Deno.test(`test ${i}`, () => {});
}
```

```
~/p/g/d/deno ❯❯❯ hyperfine --warmup=3 -- "deno test --no-check -A test.ts"  "./target/release/deno test --no-check -A test.ts"
Benchmark 1: deno test --no-check -A test.ts
  Time (mean ± σ):     587.4 ms ±   2.8 ms    [User: 594.9 ms, System: 131.3 ms]
  Range (min … max):   584.2 ms … 593.1 ms    10 runs

Benchmark 2: ./target/release/deno test --no-check -A test.ts
  Time (mean ± σ):     288.1 ms ±   2.5 ms    [User: 280.7 ms, System: 105.0 ms]
  Range (min … max):   286.6 ms … 295.1 ms    10 runs

  Warning: Statistical outliers were detected. Consider re-running this benchmark on a quiet PC without any interferences from other programs. It might help to use the '--warmup' or '--prepare' options.

Summary
  './target/release/deno test --no-check -A test.ts' ran
    2.04 ± 0.02 times faster than 'deno test --no-check -A test.ts'
```

# `Deno.test` with `await t.step()`

```ts
for (let i = 0; i < 100; i++) {
  Deno.test(`test ${i}`, async (t) => {
    for (let j = 0; j < 100; j++) {
      await t.step(`step ${j}`, async () => {});
    }
  });
}
```

```
~/p/g/d/deno ❯❯❯ hyperfine --warmup=3 -- "deno test --no-check -A test.ts"  "./target/release/deno test --no-check -A test.ts"
Benchmark 1: deno test --no-check -A test.ts
  Time (mean ± σ):      1.085 s ±  0.018 s    [User: 0.987 s, System: 0.310 s]
  Range (min … max):    1.071 s …  1.128 s    10 runs

Benchmark 2: ./target/release/deno test --no-check -A test.ts
  Time (mean ± σ):     524.4 ms ±   3.1 ms    [User: 399.1 ms, System: 238.2 ms]
  Range (min … max):   518.9 ms … 527.8 ms    10 runs

Summary
  './target/release/deno test --no-check -A test.ts' ran
    2.07 ± 0.04 times faster than 'deno test --no-check -A test.ts'
```

### CODE REVIEW

File: runtime/ops/worker_host.rs [hunk 1]
```diff
@@ -154,10 +154,8 @@ fn op_create_worker(
   };
   let parent_permissions = parent_permissions.clone();
   let worker_id = state.take::<WorkerId>();
-  let create_web_worker_cb = state.take::<CreateWebWorkerCbHolder>();
-  state.put::<CreateWebWorkerCbHolder>(create_web_worker_cb.clone());
-  let format_js_error_fn = state.take::<FormatJsErrorFnHolder>();
-  state.put::<FormatJsErrorFnHolder>(format_js_error_fn.clone());
+  let create_web_worker_cb = state.borrow::<CreateWebWorkerCbHolder>().clone();
```
Comment (lucacasonato) [inline runtime/ops/worker_host.rs:157]: Drive-by cleanup

**Review by bartlomieju — APPROVED** [review]
LGTM, great speed up

File: cli/ops/testing.rs [hunk 2]
```diff
@@ -123,92 +117,128 @@ struct TestRegisterResult {
...
-  parent_id: usize,
-  root_id: usize,
-  root_name: String,
+  ret_buf.copy_from_slice(&(id as u32).to_le_bytes());
+  Ok(origin)
 }
 
-#[op2]
-#[serde]
+#[op2(fast)]
```
Comment (ry) [inline cli/ops/testing.rs:162]: Awesome to see fast ops being used in more places
